File: Leetcode/RomanToInteger.cs
public class Solution
{
    public int RomanToInt(string s)
    {
        var romanMap = new Dictionary<char, int>() {
            { 'I', 1 }, { 'V', 5 }, { 'X', 10 }, { 'L', 50 }, { 'C', 100 }, { 'D', 500 }, { 'M', 1000 } };
        var romanInt = 0;

        for (int i = 0; i < s.Length; i++)
            if (i + 1 < s.Length && romanMap[s[i]] < romanMap[s[i + 1]])
                romanInt -= romanMap[s[i]];
            else
                romanInt += romanMap[s[i]];

        return romanInt;
    }
}

File: Leetcode/RomanToInteger.py
# Given a roman numeral, convert it to an integer.

class Solution(object):
    def romanToInt(s):
        roman_map = {"I": 1, "V": 5, "X": 10,
                    "L": 50, "C": 100, "D": 500, "M": 1000}
        roman_int = 0

        for i in range(len(s)):
            if i + 1 < len(s) and roman_map[s[i]] < roman_map[s[i + 1]]:
                roman_int -= roman_map[s[i]]
            else:
                roman_int += roman_map[s[i]]

        return roman_int